holoviews/element/selection.py

```diff
@@ -36,6 +36,47 @@ def _get_selection_expr_for_stream_value(self, **kwargs):
     def _merge_regions(region1, region2, operation):
         return None
 
+def spatial_select_gridded(xvals, yvals, geometry):
+    rectilinear = (np.diff(xvals, axis=0) == 0).all()
+    if rectilinear:
+        from .raster import Image
+        from .path import Polygons
+        try:
+            from ..operation.datashader import rasterize
+        except ImportError:
+            raise ImportError("Lasso selection on gridded data requires "
+                              "datashader to be available.")
```

Comment:
Nice use of Datashader for this purpose! Can you not fall through to the spatial_select_columnar case when datashader is not available?

Reply:
Sure, can do, although for large rasters this will be slow.

Reply:
Right.  I'm not sure if it makes sense to warn in such a case; on the one hand without a warning people might not realize that there's an easy way to make it go much faster, but on the other hand, if people genuinely can't install datashader or don't want to or have a small dataset, having such a warning will be annoying.

Reply:
Tbh, spatialpandas/shapely are more difficult to install and this isn't worth the extra complexity and possible user confusion.